android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BroadcastActivity.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.physical_web.physicalweb;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothManager;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.util.Log;
+import android.util.Patterns;
+import android.widget.Toast;
+
+import org.physical_web.physicalweb.ble.UriBeacon;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+
+
+/**
+ * This is the main entry point for the broadcasting service.
+ * Handles the share intent.
+ */
+public class BroadcastActivity extends Activity {
+
+    private static final String TAG  = "BroadcastActivity";
+    private static final int ENABLE_BLUETOOTH_REQUEST_ID = 1;
+    public static final int MAX_URI_LENGTH = 18;
+    private String url;
```

Comment:
Does this still need to be a member?  It appears to be unused except in parseAndHandleUrl where the value is set.